### PR TITLE
Add porter mixin create command

### LINKED ISSUE #1512: Add porter mixin create command
It would be great to have a command that helps you clone and get started with skeletor.

### PR DESCRIPTION
**Pull request by joshuabezaleel**
# What does this change
Introduce `porter mixin create` command which clone the [`getporter/skeletor`](https://github.com/getporter/skeletor) repo and modify the skeletor's file's content according to the argument and flags supplied by the user when running the command according to the [skeletor's README instruction](https://github.com/getporter/skeletor#a-porter-mixin-skeleton).

The structure of the command is as follows as @carolynvs mentioned [here](https://github.com/getporter/porter/issues/1512#issuecomment-811989386).
```porter mixin create NAME --author YOURNAME [--dir /path/to/mixin/dir]```

# What issue does it fix
Closes #1512 

# Notes for the reviewer
I attached a short video demo of running the command below.

https://user-images.githubusercontent.com/7043511/119260541-73de4a80-bbfd-11eb-893a-c624b2c6f5f4.mp4

# Checklist
- [x] Unit Tests
- [x] Documentation
- [x] Schema (porter.yaml)

### CODE REVIEW

**Comment by joshuabezaleel** [comment]
Hi @carolynvs , I'm truly sorry for a really late PR. Looking forward to the review! 🙂

File: pkg/porter/mixins.go [hunk 1]
```diff
@@ -106,3 +117,93 @@ func (p *Porter) GenerateMixinFeed(opts feed.GenerateOptions) error {
...
+		return err
+	}
+
+	err = os.Rename(skeletorDestPath+"/pkg/skeletor", skeletorDestPath+"/pkg/"+opts.MixinName)
+	if err != nil {
+		return err
+	}
+
+	replacementList := map[string]string{
+		"get.porter.sh/mixin/skeletor":       fmt.Sprintf("github.com/%s/%s", opts.AuthorName, opts.MixinName),
```
Comment (carolynvs) [inline pkg/porter/mixins.go:172]: We will need one more flag to collect the user's github username. --author has their full name, e.g. "Carolyn Van Slyck", which won't work as a replacement for the package name. How about `--username`?

File: cmd/porter/mixins.go [hunk 2]
```diff
@@ -197,3 +198,32 @@ func BuildMixinFeedTemplateCommand(p *porter.Porter) *cobra.Command {
...
+	cmd := &cobra.Command{
+		Use:   "create NAME --author YOURNAME [--dir /path/to/mixin/dir]",
+		Short: "Create a new mixin project based on the getporter/skeletor repository",
+		Long: `Create a new mixin project based on the getporter/skeletor repository.
+
+The first argument is the name of the mixin to create and is required.
+A flag of --author to declare the author of the mixin is also a required input.
+You can also specify where to put mixin directory. It will default to the current directory.`,
+		Example: ` porter mixin create MyMixin --author MyName
+		porter mixin create MyMixin --author MyName --dir path/to/mymixin
```
Comment (carolynvs) [inline cmd/porter/mixins.go:214]:
```suggestion
		porter mixin create MyMixin --author "My Name" --username myuser --dir path/to/mymixin
```

Let's show them how to use a full name with spaces, and then include a new flag for the username (more on that below).

**Review by carolynvs — CHANGES_REQUESTED** [review]
Wonderful! 🙏🏼 Thank you for putting this together. There are just two things that I'd like to include in this pull request as well:

* Mention the command and how to use it on our website so that people know about it. https://porter.sh/mixin-dev-guide/ right now points people to the skeletor repo. Can you add a quick sentence or two and a example command showing someone how to use `porter mixins create`?
* This doesn't have any tests and it would be easy for us to miss if anything broke. Can you write a test that does the following?
    1. Run the create command in a temp directory. You can mimic a test like [this one](https://github.com/getporter/porter/blob/f7cdf95987a8523b85aa8bb7a88dd000ea6541fb/pkg/porter/mixins_test.go#L13) to do it
    2. Run `make build` using exec.Command in that directory and check that it doesn't return an error.

**Comment by carolynvs** [comment]
@joshuabezaleel Just checking in to see if you had any questions about the review feedback, or were having trouble implementing the suggestions. I'm happy to help get this PR finished up!

**Comment by joshuabezaleel** [comment]
Hi @carolynvs , I am really sorry for taking a really long time to make an update on this PR. Life got hard for a while, some personal issues and the COVID situation here in Indonesia turns to be really overwhelming and gets the best of me, making me not to have any more energy outside work especially on open source.

It's getting better and I think I am ready to get back to contributing to Porter. Thank you lots for the great review, I totally skipped about it while implementing and your points on flag `--username` and adding unit test is spot on. I've added the flag, test, and also description that you might want to look for its wording on the docs since English isn't my first and day-to-day language. 

Looking forward to further review, and many more Porter issues to tackle 💪 Hope you all Porter folks are also doing well in these times ❤️

**Comment by carolynvs** [comment]
@joshuabezaleel Looks like this is ready for another review! I'll try to get to this today.

Only one of your commits has a line that says "signed-off-by ... Josh", which is why the DCO check is failing. If you are comfortable with git, you can fix that yourself with `git rebase release/v1 --signoff`. If you run into trouble [signing your commits](https://github.com/getporter/porter/blob/main/CONTRIBUTING.md#signing-your-commits), leave a comment here that it's okay for me to squash your commits into a single one and use the signoff from your first commit.

Also there is a merge conflict on one of the files. If you can fix that, great, if not I'll resolve it when I merge your PR.

**Comment by joshuabezaleel** [comment]
Hi @carolynvs , sorry for the late update! Thank you so much for the really helpful inputs on the review and sorry for the unintentional changes that I've made on the last PR. I've tried to rebase the branch, resolve the conflict, and sign-off all of my commits, hopefully it doesn't cause any problems. 😅

**Review by carolynvs — APPROVED** [review]
Looks great, thanks!

I have added a commit to your PR that removes some files that should not have been checked into the tests/smoke directory.

**Comment by carolynvs** [comment]
This will go out in the next alpha release of Porter! 🎉

**Pull request merged**